### PR TITLE
Reconfigure pylint and fix or exempt new warnings

### PR DESCRIPTION
**Pull request by lukpueh**
This PR 
- moves pylint config from pylintrc to pyproject.toml, which is state of the art
- adopts a more succinct and restrictive pylint configuration (as used in python-tuf and securesystemslib)
- fixes new warnings, if feasible, i.e. changes are minimal
- otherwise disables warnings as locally as possible, to appease the linter for existing issues, but allow it to flag issues in new code

Although the diff is big the changes are very repetitive and grouped in commits per linter warning.

### CODE REVIEW

File: tests/__init__.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
 # Copyright New York University and the in-toto contributors
 # SPDX-License-Identifier: Apache-2.0
+"""Test in-toto. """
```
Comment (adityasaky) [inline tests/__init__.py:3]:
```suggestion
"""Test in-toto."""
```

File: tests/test_in_toto_match_products.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+"""Test in_toto_match_products.py """
```
Comment (adityasaky) [inline tests/test_in_toto_match_products.py]:
```suggestion
"""Test in_toto_match_products.py."""
```

**Review by adityasaky — APPROVED** [review]
LGTM, nits. Should module level docstrings have a trailing space?

File: tests/__init__.py (hunk 1 above)
Comment (lukpueh) [reply tests/__init__.py:3]: Yes, black does this.

File: tests/test_in_toto_match_products.py (hunk 2 above)
Comment (lukpueh) [reply tests/test_in_toto_match_products.py]: I think `pydocstyle` would check this.

**Pull request merged**